### ISSUE TITLE
Installation via Powershell Core / .NET Core broken/unsupported due to dependency on Attribute-based versioning

### ISSUE THREAD

**Issue by HEskandari** [Particular/ServiceControl#2123]
### Problem Description

Result of some investigation shows that the new Powershell versions that are .NET Core based do work with most of our cmdlets but there are a few that are broken.

The broken ones are the ones that do the installations (e.g. `New-ServiceControlInstance`) and that is due to how we store release date information as an assembly attribute. The .NET Core runtime seem to have issues reading the values back which stops those cmdlets working properly.

### Suggested Solution

Put the release date information as a text file in the release artifacts (zip files). Read the information from the zip file at installation time.

**Comment by mauroservienti** [comment]
I did an investigation on this today. The installation process checks if the license is valid at installation time, and to do so it needs to extract the release date from the assembly it's trying to install.

The release date is stored in assemblies by the Licensing package. The Licensing package at build time, using an MSBuild task, [writes a `ReleaseDate` attribute](https://github.com/Particular/Operations.Licensing/blob/13608e31ab7ce46e357fd777380b9bc57051ab1b/src/Particular.Licensing/Particular.Licensing.Sources.targets#L16-L18) to the compiled assembly storing the GitVersion `$(GitVersion_CommitDate)`.

Finally, `ServiceControlInstaller.Packaging` project [zips all assemblies](https://github.com/Particular/ServiceControl/blob/a56860be685270d06490342e90007e84480bf00e/src/ServiceControlInstaller.Packaging/ServiceControlInstaller.Packaging.csproj#L38) at build time using an MSBuild task.

At installation time the release date is read by the engine from using reflection from the ServiceControl executable: https://github.com/Particular/ServiceControl/blob/a56860be685270d06490342e90007e84480bf00e/src/ServiceControlInstaller.Engine/FileSystem/ReleaseDateReader.cs#L33-L46

As suggested by @HEskandari one option here is to:

- make so that the build process stores the release date in the zip file using a text file
- tweak the above code to read from the text file in case of failure

Another option that requires fewer changes is to use `Mono.Cecil` to read the assembly definition. I just tested it by reading a custom attribute defined in a .NET Framework 4.8 assembly from a .NET Core 3.1 console application. It's as simple as:

```csharp
var assemblyPath = @"path to the assembly to read";
var assemblyDefinition = Mono.Cecil.AssemblyDefinition.ReadAssembly(assemblyPath);

var customAttribute = assemblyDefinition.CustomAttributes.SingleOrDefault(ca => ca.AttributeType.Name == "CustomAttribute");
var constructorArgument = customAttribute?.ConstructorArguments[0];
```

Raised https://github.com/Particular/ServiceControl/pull/2314 to show how it can be used in ServiceControl.
